docs/docs/language_concepts/01_functions.md

```diff
@@ -103,3 +103,45 @@ assert(add_50(100) == 150);
 ```
 
 See [Lambdas](./08_lambdas.md) for more details.
+
+## Attributes
+
+Attribute is metadata that can be applied to a function, using the following syntax: `#[attribute(value)]`.
+
+Supported attributes include:
+- **builtin**: the function is implemented by the compiler, for efficiency purposes.
+- **deprecated**: mark the function as *deprecated*. Calling the function will generate a warning: `warning: use of deprecated function`
+- **field**: conditional compilation of code. See below for more details
+- **oracle**: mark the function as *oracle*; meaning it is an external unconstrained function, implemented in noir_js. See [Unconstrained](./05_unconstrained.md) and [Noir js](../noir_js/noir_js.md) for more details.
+- **test**: mark the function as unit tests. See [Tests](../nargo/02_testing.md) for more details
```

Comment:
Can we re-order this attribute section? I think it'd be better if attributes were ordered by their importance to end users. E.g. 1. test, 2. field, 3. deprecated, 4. builtin, and 5. oracle